File: Composicion y Agregacion/ejercicio 1/Java/Habitacion.java
import java.util.ArrayList;
class Habitacionn {
    public String nombre;
    public double tamano;
    public Habitacionn(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }
    public String getNombre() {
        return nombre;
    }
    public double getTamano() {
        return tamano;
    }
    public void mostrar() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m2");
    }
}
class Casa {
    public String direccion;
    public ArrayList<Habitacionn> habitaciones;
    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }
    public String getDireccion() {
        return direccion;
    }
    public void agregar(Habitacionn habitacion) {
        habitaciones.add(habitacion);
    }
    public void mostrar() {
        System.out.println("Casa en: " + direccion);
        for (Habitacionn h : habitaciones) {
            h.mostrar();
        }
    }
}
public class Habitacion {
    public static void main(String[] args) {
        Casa c = new Casa("Calle Falsa 123");
        c.agregar(new Habitacionn("Sala", 20));
        c.agregar(new Habitacionn("Cocina", 10));
        c.agregar(new Habitacionn("Dormitorio", 15));
        c.mostrar();
    }
}

File: Composicion y Agregacion/ejercicio 1/Python/Habitacion.py
class Habitacion:
    def __init__(self, nombre, tamano):
        self.nombre = nombre
        self.tamano = tamano
    def mostrar(self):
        print("Habitacion:", self.nombre)
        print("Tamaño:", self.tamano, "m^2")
class Casa:
    def __init__(self, direccion):
        self.direccion = direccion
        self.habitaciones = []
    def agregar(self, habitacion):
        self.habitaciones.append(habitacion)
    def mostrar(self):
        print("Direccion:", self.direccion)
        for h in self.habitaciones:
            h.mostrar()
h1 = Habitacion("Sala", 25)
h2 = Habitacion("Cocina", 15)
c = Casa("Calle 123")
c.agregar(h1)
c.agregar(h2)
c.mostrar()
